src/Glpi/Api/Rest/Client.php

```diff
@@ -153,13 +155,14 @@ public function request($method, $uri, array $options = []) {
          $hasResponse = $e->hasResponse();
          $statusCode = ($hasResponse) ? $e->getResponse()->getStatusCode() : '500';
          $contents = ($hasResponse) ? $e->getResponse()->getReasonPhrase() : 'Request Error';
-         throw new Exception($contents, $statusCode);
+         throw new BadEndpointException($contents, $statusCode);
```

Comment:
the phpdoc is wrong : this method returns a \Psr\Http\Message\ResponseInterface and nothing else.

If it does not returns an isntance of \Psr\Http\Message\ResponseInterface the method throws an exception. No mixed not null